examples/levels/src/reverse.ts

```diff
@@ -0,0 +1,225 @@
+import {
+    ManifestClient,
+    WrapperPlaceOrderReverseParamsExternal,
+} from '@cks-systems/manifest-sdk';
+import { OrderType } from '@cks-systems/manifest-sdk/client/ts/src/manifest/types';
+import { 
+    Connection, 
+    Keypair, 
+    Commitment, 
+    sendAndConfirmTransaction, 
+    Transaction, 
+    PublicKey
+} from "@solana/web3.js";
+import * as dotenv from "dotenv";
+import bs58 from "bs58";
+import { 
+    INITIAL_FDV,
+    INITIAL_QTY_RATIO, 
+    INITIAL_QTY_REDUCTION,
+    QTY_DECR_REDUCTION,
+    INITIAL_PRICE_INCR,
+    PRICE_INCR_REDUCTION,
+    TOKEN_SUPPLY,
+    NUM_LEVELS,
+    MAX_NUM_ORDER_IX
+} from './constants';
+
+dotenv.config();
+
+// Envs
+const PRIVATE_KEY = process.env.PRIVATE_KEY;
+const RPC_URL = process.env.RPC_URL;
+const MFX_MKT = process.env.MFX_MKT;
+
+interface BondingLevel {
+    price: number;
+    quantity: number;
+    level: number;
+}
+
+/**
+ * Generates order levels similar to a bonding curve from AMMs 
+ * Returns the bonding levels corresponding prices and quantities.
+ * Price increases and quantity decreases are reduced at each level.
+ */
+function generateBondingLevels(
+    totalSupply: number = 1_000_000_000,
+    numLevels: number = 100,
+): BondingLevel[] {
+    const levels: BondingLevel[] = [];
+    const initialPrice = INITIAL_FDV / totalSupply;
+    
+    let currentQuantity = totalSupply * INITIAL_QTY_RATIO;
+    let currentPrice = initialPrice;
+    
+    for (let level = 0; level < numLevels; level++) {
+        levels.push({
+            price: currentPrice,
+            quantity: currentQuantity,
+            level: level + 1,
+        });
+        const currentPriceIncrease = Math.max(INITIAL_PRICE_INCR - (level * PRICE_INCR_REDUCTION), 0);
+        const currentQuantityReduction = Math.min(0, INITIAL_QTY_REDUCTION - (level * QTY_DECR_REDUCTION));
+        
+        currentQuantity *= (1 + currentQuantityReduction);
+        currentPrice *= (1 + currentPriceIncrease);
+    }
+
+    console.log('Price Range:', {
+        initialPrice,
+        finalPrice: levels[levels.length - 1].price,
+        priceMultiple: levels[levels.length - 1].price / initialPrice,
+        numLevels: levels.length
+    });
+    
+    return levels;
+}
+
+/**
+ * Gets the current level based on total tokens sold.
```

Comment:
This doesnt work anymore with nonzero spreads on reverse orders

Reply:
Renamed

Reply:
If we need to get the current level because of the possibility of switching markets, the reason we have this, then wont it almost always be wrong?
And even if it was configured the same, wont spreads make it so that levels can get out of order and this will be off in the higher levels?

Reply:
it will be approximate and still serve the purpose of resetting on a new market